File: server/Models/Entities/Student.cs
using System.ComponentModel.DataAnnotations;

namespace server.Models.Entities
{
    public class Student : User
    {
        [Key]
        [Required]
        public string Id { get; set; } = string.Empty;

        [Required]
        public string? StudentNumber { get; set; }

        public Guid? GroupId { get; set; }

        public virtual ResearchGroup? ResearchGroup { get; set; }

        [Required]
        [MaxLength(50)]
        public string? Position { get; set; }

        [Required]
        [MaxLength(100)]
        public string? Institute { get; set; }

        [Required]
        [MaxLength(100)]
        public string? Program { get; set; }

        [Required]
        public int? YearLevel { get; set; }

        [Required]
        [MaxLength(1)]
        public char Section { get; set; }
    }
}

File: packages/types/src/student.ts
import type { ResearchGroupResponseDto } from "./research-group";

export interface CreateStudentDto {
  id: string;
  email: string;
  firstName: string;
  middleInitial?: string;
  lastName: string;
  suffix?: string;
  studentNumber: string;
  groupId?: string;
  position: string;
  institute: string;
  program: string;
  yearLevel: number;
  section: string;
}

export interface UpdateStudentDto {
  email?: string;
  firstName?: string;
  middleInitial?: string;
  lastName?: string;
  suffix?: string;
  studentNumber?: string;
  groupId?: string;
  position?: string;
  institute?: string;
  program?: string;
  yearLevel?: number;
  section?: string;
  isActive?: boolean;
}

export interface StudentResponseDto {
  id: string;
  email: string;
  firstName: string;
  middleInitial?: string;
  lastName: string;
  suffix?: string;
  studentNumber: string;
  researchGroup?: ResearchGroupResponseDto;
  position: string;
  institute: string;
  program: string;
  yearLevel: number;
  section: string;
  role: string;
  isActive?: boolean;
  createdAt: string;
  updatedAt: string;
}

export type StudentResponseListDto = StudentResponseDto[];